File: C#/BluffinMuffin.Protocol.DataTypes/TableParams.cs
using BluffinMuffin.Protocol.DataTypes.Attributes;
using BluffinMuffin.Protocol.DataTypes.Json;
using BluffinMuffin.Protocol.DataTypes.Options;
using Newtonsoft.Json;
using BluffinMuffin.Protocol.DataTypes.Enums;

namespace BluffinMuffin.Protocol.DataTypes
{
    /// <summary>
    /// 
    /// </summary>
    public class TableParams
    {       
        /// <summary>
        /// The name of the table
        /// </summary>
        [ExampleValue("Bikini Bottom")]
        public string TableName { get; set; }

        /// <summary>
        /// The variant of the GameType
        /// </summary>
        [ExampleValue(GameSubTypeEnum.TexasHoldem)]
        public GameSubTypeEnum Variant { get; set; }

        /// <summary>
        /// The minimum seated players needed to start the game. (Between 2 and 10)
        /// </summary>
        [ExampleValue(2)]
        public int MinPlayersToStart { get; set; }

        /// <summary>
        /// The maximum number of players. (Between 2 and MaxPlayers of the GameInfo, and must be >= MinPlayersToStart)
        /// </summary>
        [ExampleValue(10)]
        public int MaxPlayers { get; set; }

        /// <summary>
        /// The waiting times (At different stage of the game, the server will wait before continuing to making it feel real !)
        /// </summary>
        public ConfigurableWaitingTimes WaitingTimes { get; set; }

        /// <summary>
        /// The size of the table. It represents the small bet and the big blind. Big bet will be double the GameSize, small blind is half the GameSize, and Ante is 1/10 the GameSize
        /// </summary>
        [ExampleValue(10)]
        public int GameSize { get; set; }

        /// <summary>
        /// Some more parameters. Those parameters are specific to a server and are free-format. Generally, this field would be blank.
        /// </summary>
        [ExampleValue("")]
        public string Arguments { get; set; }

        /// <summary>
        /// The type of blinds the table uses (none, blinds, antes)
        /// </summary>
        [ExampleValue(BlindTypeEnum.Blinds)]
        public BlindTypeEnum Blind { get; set; }

        /// <summary>
        /// The type of limit the table uses (NoLimit, PotLimit, FixedLimit)
        /// </summary>
        [ExampleValue(LimitTypeEnum.NoLimit)]
        public LimitTypeEnum Limit { get; set; }

        /// <summary>
        /// The type of table it is (QuickMode ? RegisteredMode ?) See 'BluffinMuffin.Protocol.DataTypes.Options.LobbyOptions' for more details
        /// </summary>
        [ExampleValue(typeof(GameTypeOptionsCommunity))]
        [JsonConverter(typeof(OptionJsonConverter<GameTypeOptions, GameTypeEnum>))]
        [JsonProperty(Order = 100)]
        public GameTypeOptions Options { get; set; }

        /// <summary>
        /// The type of table it is (QuickMode ? RegisteredMode ?) See 'BluffinMuffin.Protocol.DataTypes.Options.GameTypeOptions' for more details
        /// </summary>
        [ExampleValue(typeof(LobbyOptionsQuickMode))]
        [JsonConverter(typeof(OptionJsonConverter<LobbyOptions, LobbyTypeEnum>))]
        [JsonProperty(Order = 100)]
        public LobbyOptions Lobby { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public TableParams()
        {
            Arguments = string.Empty;
            TableName = "Anonymous Table";
            Variant = GameSubTypeEnum.TexasHoldem;
            MinPlayersToStart = 2;
            MaxPlayers = 10;
            WaitingTimes = new ConfigurableWaitingTimes();
            GameSize = 10;
            Lobby = new LobbyOptionsQuickMode();
            Blind = BlindTypeEnum.Blinds;
            Limit = LimitTypeEnum.NoLimit;
            Options = new GameTypeOptionsCommunity();
        }

        /// <summary>
        /// MinimumBuyInAmount
        /// </summary>
        [JsonIgnore]
        public int MinimumBuyInAmount
        {
            get
            {
                switch (Lobby.MinimumBuyInParameter)
                {
                    case BuyInParameterEnum.FixedAmount:
                        return Lobby.MinimumBuyInValue;
                    case BuyInParameterEnum.Multiplicator:
                        return Lobby.MinimumBuyInValue*GameSize;
                }
                return 0;
            }
        }

        /// <summary>
        /// MaximumBuyInAmount
        /// </summary>
        [JsonIgnore]
        public int MaximumBuyInAmount
        {
            get
            {
                switch (Lobby.MaximumBuyInParameter)
                {
                    case BuyInParameterEnum.FixedAmount:
                        return Lobby.MaximumBuyInValue;
                    case BuyInParameterEnum.Multiplicator:
                        return Lobby.MaximumBuyInValue * GameSize;
                }
                return int.MaxValue;
            }
        }
    }
}

File: Python/bluffinmuffin/protocol/data_types/table_params.py
from collections import OrderedDict

from bluffinmuffin.protocol.enums import GameSubTypeEnum
from bluffinmuffin.protocol.enums import BlindTypeEnum
from bluffinmuffin.protocol.enums import LimitTypeEnum
from .configurable_waiting_times import ConfigurableWaitingTimes
from .lobby_options import LobbyOptionsDecoder
from .game_type_options import GameTypeOptionsDecoder


class TableParams:
    def __init__(self, table_name, variant, min_players_to_start, max_players, waiting_times, game_size,
                 lobby, blind, limit, arguments, options):
        self.table_name = table_name
        self.variant = variant
        self.min_players_to_start = min_players_to_start
        self.max_players = max_players
        self.waiting_times = waiting_times
        self.game_size = game_size
        self.arguments = arguments
        self.lobby = lobby
        self.blind = blind
        self.limit = limit
        self.arguments = arguments
        self.options = options

    def __str__(self):
        return '"{0}", {1}, {2}/{3}, {5}, {6}, {7}, {8}, {9}, {10}'.format(
            self.table_name,
            GameSubTypeEnum.to_string(self.variant),
            self.min_players_to_start,
            self.max_players,
            self.waiting_times,
            self.game_size,
            self.arguments,
            self.lobby,
            BlindTypeEnum.to_string(self.blind),
            LimitTypeEnum.to_string(self.limit),
            self.options
        )

    def encode(self):
        d = OrderedDict()
        d['TableName'] = self.table_name
        d['Variant'] = GameSubTypeEnum.to_string(self.variant)
        d['MinPlayersToStart'] = self.min_players_to_start
        d['MaxPlayers'] = self.max_players
        d['WaitingTimes'] = self.waiting_times.encode()
        d['GameSize'] = self.game_size
        d['Lobby'] = self.lobby.encode()
        d['Blind'] = BlindTypeEnum.to_string(self.blind)
        d['Limit'] = LimitTypeEnum.to_string(self.limit)
        d['Arguments'] = self.arguments
        d['Options'] = self.options.encode()
        return d

    @classmethod
    def decode(cls, obj):
        return cls(
            obj["TableName"],
            GameSubTypeEnum.parse(obj['Variant']),
            obj["MinPlayersToStart"],
            obj["MaxPlayers"],
            ConfigurableWaitingTimes.decode(obj['WaitingTimes']),
            obj["GameSize"],
            LobbyOptionsDecoder.decode(obj['Lobby']),
            BlindTypeEnum.parse(obj['Blind']),
            LimitTypeEnum.parse(obj['Limit']),
            obj["Arguments"],
            GameTypeOptionsDecoder.decode(obj['Options']),
        )
